[TT-61] Split output into two

emmiegit opened the pull request:
As described in [TT-61](https://scuttle.atlassian.net/browse/TT-61), due to Wikidot page size limits, we now have two separate pages with outputted tag data. There is the [Tech Hub Tag List](https://05command.wikidot.com/tech-hub-tag-list) which has the generated hub as you'd expect, and now there is also [Tech Hub Tag List Crom Search](https://05command.wikidot.com/tech-hub-tag-list-crom-search) which only contains the TOML data used by [Crom Search](https://scpwiki.com/search:crom) to give tag recommendations. (The [Tag List Manifest](https://05command.wikidot.com/tag-list-manifest) remains in the same place.)

This PR changes the tag-list generator to split out the outputs, with two input templates, two sets of input and output boxes, and refactoring the code appropriately so it can handle two or more templates structurally.

![Screenshot of tag-list after changes](https://user-images.githubusercontent.com/8848022/222365379-3ca6fc1c-ef88-4c85-b691-1ed73876332c.png)

This also makes some other changes, such as the use of `scpwiki.com` instead of `scp-wiki.wikidot.com` in generated tag URLs, ~~and updating those large tag TOML files in the repositories, since it's been a while. (Question, though, why are those there? Is it an artifact of development and can be removed now?)~~

[TT-61]: https://scuttle.atlassian.net/browse/TT-61?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

emmiegit committed:
Use scpwiki.com domain.

emmiegit committed:
Use HTTPS for O5.

emmiegit committed:
Add development instructions.

emmiegit committed:
Update committed tag TOML data.

I'm not entirely sure why this is in the repo, but because it's here I'm
going to update the data since it's been a while and a lot has changed.

emmiegit committed:
Create separate output templates.

emmiegit committed:
Use scpwiki.com in generation.

emmiegit committed:
Update category names.

emmiegit committed:
Add missing space.

emmiegit committed:
Rename template files to have the same prefix.

emmiegit committed:
Add separate input for second template.

emmiegit committed:
Rename toml to data.

emmiegit committed:
Delete "other".

Folded into other groups, mostly "events".

emmiegit committed:
Split templates into two sections in index.ts

emmiegit committed:
Add separate output fields.

emmiegit committed:
Split makeOutput() into separate calls.

Rename main makeOutput() -> makeOutputs(), then have a new
inner implementation function makeOutput() which outputs for one
template.

emmiegit committed:
Track both templates.

emmiegit committed:
Remove spaces from after function names.

rossjrw reviewed (approved):
Looks good overall to me.

One concern is that this effectively hardcodes the EN usage of this tool, though it's possible that other branches are using it. I don't know that for sure though, especially given that it's written in English. Honestly at this stage I'm happy to canonise this as an EN tool that would need to be forked and modified to be used by another branch. Maybe in the future the number of text boxes could be configured by URL parameter like some of the other settings.

The TOML files in the repo are absolutely an artifact of development. They can be deleted.

emmiegit committed:
Remove tag TOML files used in development.

They live in O5 now, not here.

emmiegit commented:
> Looks good overall to me.
> 
> One concern is that this effectively hardcodes the EN usage of this tool, though it's possible that other branches are using it. I don't know that for sure though, especially given that it's written in English. Honestly at this stage I'm happy to canonise this as an EN tool that would need to be forked and modified to be used by another branch. Maybe in the future the number of text boxes could be configured by URL parameter like some of the other settings.
> 
> The TOML files in the repo are absolutely an artifact of development. They can be deleted.

I agree it's a bit EN-specific, but it's been a bit and I don't think other branches are using this tool specifically. I've been of a mind to target this at the existing use case and adapt as needed, while still being flexible for potential futures.

Regarding those embedded TOML files, since they were only used for development, I have removed them.

emmiegit commented:
thanks @rossjrw
